packages/framework-core/src/services/read-model-store.ts

```diff
@@ -35,42 +35,52 @@ export class ReadModelStore {
     }
     const entityMetadata = this.config.entities[entitySnapshotEnvelope.entityTypeName]
     await Promises.allSettledAndFulfilled(
-      projections.map(async (projectionMetadata: ProjectionMetadata) => {
+      projections.flatMap((projectionMetadata: ProjectionMetadata) => {
         const readModelName = projectionMetadata.class.name
         const entityInstance = createInstance(entityMetadata.class, entitySnapshotEnvelope.value)
-        const readModelID = this.joinKeyForProjection(entityInstance, projectionMetadata)
+        const readModelIDList = this.joinKeyForProjection(entityInstance, projectionMetadata)
         const sequenceKey = this.sequenceKeyForProjection(entityInstance, projectionMetadata)
-        if (!readModelID) {
+        if (!readModelIDList) {
           this.logger.warn(
             `Couldn't find the joinKey named ${projectionMetadata.joinKey} in entity snapshot of ${entityMetadata.class.name}. Skipping...`
           )
           return
         }
-        this.logger.debug(
-          '[ReadModelStore#project] Projecting entity snapshot ',
-          entitySnapshotEnvelope,
-          ` to build new state of read model ${readModelName} with ID ${readModelID}`,
-          sequenceKey ? ` sequencing by ${sequenceKey.name} with value ${sequenceKey.value}` : ''
-        )
 
-        return await retryIfError(
-          this.logger,
-          () =>
-            this.applyProjectionToReadModel(
-              entityInstance,
-              projectionMetadata,
-              readModelName,
-              readModelID,
-              sequenceKey
-            ),
-          OptimisticConcurrencyUnexpectedVersionError
-        )
+        return readModelIDList.map(async (readModelID: UUID) => {
+          this.logger.debug(
+            '[ReadModelStore#project] Projecting entity snapshot ',
+            entitySnapshotEnvelope,
+            ` to build new state of read model ${readModelName} with ID ${readModelID}`,
+            sequenceKey ? ` sequencing by ${sequenceKey.name} with value ${sequenceKey.value}` : ''
+          )
+
+          return await retryIfError(
+            this.logger,
+            () =>
+              this.applyProjectionToReadModel(
+                entityInstance,
+                projectionMetadata,
+                readModelName,
+                readModelID,
+                sequenceKey
+              ),
+            OptimisticConcurrencyUnexpectedVersionError
+          )
+        })
```

Comment:
I **think** that `async`/`await` has no effect in this code, as the `map` operation does not await the callback function. As you're flattening the results and feeding them to the `Promise.allSettledAndFulfilled`, it's safe to remove them.

```suggestion
        return readModelIDList.map((readModelID: UUID) => {
          this.logger.debug(
            '[ReadModelStore#project] Projecting entity snapshot ',
            entitySnapshotEnvelope,
            ` to build new state of read model ${readModelName} with ID ${readModelID}`,
            sequenceKey ? ` sequencing by ${sequenceKey.name} with value ${sequenceKey.value}` : ''
          )

          return retryIfError(
            this.logger,
            () =>
              this.applyProjectionToReadModel(
                entityInstance,
                projectionMetadata,
                readModelName,
                readModelID,
                sequenceKey
              ),
            OptimisticConcurrencyUnexpectedVersionError
          )
        })
```